src/zkevm_specs/evm/execution/extcodecopy.py

```diff
@@ -0,0 +1,60 @@
+from ..instruction import Instruction, Transition
+from ...util.param import EXTRA_GAS_COST_ACCOUNT_COLD_ACCESS
+from ...util import N_BYTES_MEMORY_ADDRESS, FQ
+from ..table import AccountFieldTag, CallContextFieldTag, CopyDataTypeTag
+
+
+def extcodecopy(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    address_word, memory_offset_word, code_offset_word, size_word = (
+        instruction.stack_pop(),
+        instruction.stack_pop(),
+        instruction.stack_pop(),
+        instruction.stack_pop(),
+    )
+
+    address = instruction.rlc_to_fq(address_word, 20)
+    memory_offset, size = instruction.memory_offset_and_length(memory_offset_word, size_word)
+    code_offset = instruction.rlc_to_fq(code_offset_word, N_BYTES_MEMORY_ADDRESS)
+
+    tx_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
+    is_warm = instruction.add_account_to_access_list(tx_id, address, instruction.reversion_info())
+
+    code_hash = instruction.account_read(address, AccountFieldTag.CodeHash)
+    code_size = instruction.bytecode_length(code_hash)
```

Comment:
That seems correct to me. (Responding to your comment on the PR description).
It's the bytecode table who holds the info `CodeHash` -> `Size`

Reply:
cc @ed255

Reply:
Thanks for your confirm!

Reply:
Yes! This is correct.  The prover must fill the bytecode table with all the bytecodes that will be accessed during the verification of a block.  And this includes all bytecodes read from an `EXTCODECOPY`.